### ISSUE TITLE
BMPFullscreenHandler bug

### ISSUE THREAD

**Issue by MatejBalantic** [bitmovin/bitmovin-player-ios-sdk-cocoapod#6]
I am implementing a scenario where the player is loaded inside another viewController minimized (using childViewController). That works fine, but now I would like to add support for a full-screen mode.

I noticed that on the `BMPBitmovinPlayerView` there is a `BMPFullscreenHandler`. There isn't any documentation about that, but after I implemented & assigned the handler, a new `full screen` button appeared: 

<img width="65" alt="screen shot 2018-06-23 at 10 47 00" src="https://user-images.githubusercontent.com/994179/41807700-ef3eba2a-76d2-11e8-8a03-73bffe172972.png">

After user taps that new button, the `onFullscreenRequested` on the handler is correctly invoked and my implementation then presents this player in the modal view and sets the `isFullscreen` on the handler to `true`. That's all great 👍 

Now here's the issue: in full-screen mode, after clicking that button again, `onFullscreenRequested` is called again. I would expect that with the second click the `onFullscreenExitRequested` gets called instead, but it never does. In fact, also the `isFullscreen` property on the handler is never accessed by the SDK (I validated that using breakpoints).

Is this a bug in the SDK or am I wrongly implementing the handler? Any documentation or just explanation for how the handler should work would be greatly appreciated.

**Comment by hawk23** [comment]
@MatejBalantic You've implemented the `FullscreenHandler` correctly as I can tell from your description. We will fix that `isFullscreen` and `onFullscreenExitRequested` are not called with version `2.11.0`. You can expect the release to happen latest end of next week.

**Comment by hawk23** [comment]
@MatejBalantic We just released version `2.11.0-rc.1` which contains a fix for this issue. `onFullscreenExitRequested` should be called properly now by the SDK.

**Comment by samek** [comment]
We'll test it out and let you know.

**Comment by hawk23** [comment]
@MatejBalantic @samek Did you already have time to test the fullscreen handling with `2.11.0-rc.1`?

**Comment by samek** [comment]
@hawk23 ... @MatejBalantic has to do that but he's busy organising https://addconf.com/2018/  :)

**Comment by MatejBalantic** [comment]
Hi @hawk23 , thanks for a quick fix. The handler now gets correctly called.

I've faced an issue with retain cycle through and I discovered that the SDK is strongly retaining the handler class. That can be a problem if the class happens to be the view controller that holds the PlayerView, as that creates a retain cycle.

I've worked around this by creating a separate handler class, but I would call this behavior a bug and suggest you weakly retain the handler instead.

<img width="905" alt="screen shot 2018-07-11 at 00 18 32" src="https://user-images.githubusercontent.com/994179/42541589-3774def4-84a3-11e8-9be7-496155adf6f5.png">
